app/src/main/java/com/nutomic/syncthingandroid/activities/SettingsActivity.java

```diff
@@ -512,5 +549,43 @@ protected void onPostExecute(Boolean haveRoot) {
                 }
             }
         }
+
+        /**
+         * Handles a new user input for the SOCKS proxy preference.
+         * Returns if the changed setting requires a reboot.
+         */
+        private boolean handleSocksProxyPreferenceChange(Preference preference, String newValue) {
+            // Valid input is either a proxy adress or an empty field to disable the proxy.
+            if (newValue.equals("")) {
+                preference.setSummary(getString(R.string.do_not_use_proxy) + " " + getString(R.string.generic_example) + ": " + getString(R.string.socks_proxy_address_example));
+                return true;
+            } else if (newValue.matches("^socks5://(\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}):(\\d{1,5})$")) {
```

Comment:
The ip address pattern is still invalid, this can easily be a hostname rather than IP

Reply:
Ok, i took a RFC 1123 compliant pattern from https://stackoverflow.com/questions/106179/regular-expression-to-match-dns-hostname-or-ip-address and adjusted it to our needs. Verified working on phone and online reg ex tester using different DNS, FQDN, NetBIOS and IP samples (http://www.regexplanet.com/advanced/java/index.html)